elide-contrib/elide-dynamic-config-helpers/src/main/resources/elideTableSchema.json

```diff
@@ -0,0 +1,296 @@
+            "type": "string",
+            "pattern": "^([A-Za-z]*_?[A-Za-z]*)*$",
+            "default": ""
+          },
+          "table": {
+            "title": "Table name",
+            "description": "The physical table name where the model lives.",
+            "type": "string",
+            "pattern": "^([A-Za-z]*_?[A-Za-z]*)*$",
+            "default": ""
+          },
+          "hidden": {
+            "title": "Hide/Show Table",
+            "description": "Whether this table is exposed via API metadata",
+            "type": "boolean",
+            "default": false
+          },
+          "sql": {
+            "title": "Table SQL",
+            "description": "SQL query which is used to populate the table.",
+            "type": "string",
+            "default": ""
+          },
+          "description": {
+            "title": "Table Model description",
+            "description": "A long description of the model.",
+            "type": "string",
+            "default": "[$name]"
+          },
+          "cardinality": {
+            "title": "Model cardinality",
+            "description": "The number of rows in the table: (tiny, small, medium, large, huge). The relative sizes are decided by the table designer(s).",
+            "type": "string",
+            "enum": [
+              "tiny",
+              "small",
+              "medium",
+              "large",
+              "huge"
+            ],
+            "default": "tiny"
+          },
+          "readAccess": {
+            "title": "Table read access",
+            "description": "Read permission for the table.",
+            "type": "string",
+            "default": "Allow All"
+          },
+          "joins": {
+            "title": "Table joins",
+            "description": "Describes SQL joins to other tables for column references.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/join"
+            },
+            "default": []
+          },
+          "measures": {
+            "title": "Table measures",
+            "description": "Zero or more metric definitions.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/measure"
+            },
+            "default": []
+          },
+          "dimensions": {
+            "title": "Table dimensions",
+            "description": "One or more dimension definitions.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/dimension"
+            },
+            "default": []
+          },
+          "tags": {
+            "title": "Table tags",
+            "description": "An array of string based tags",
+            "type": "array",
+            "uniqueItems": true,
+            "items": {
+              "type": "string"
+            },
+            "default": []
+          }
+        },
+        "required": [
+          "name",
+          "dimensions"
```

Comment:
one of (sql | table) is also required.

Reply:
Added this change using oneOf and also added extends